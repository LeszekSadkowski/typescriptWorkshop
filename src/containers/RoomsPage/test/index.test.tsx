import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { RoomsPage } from '../index';
import toJson, { mountToJson } from 'enzyme-to-json';

describe('rooms page container test', () => {
    it('should render given room', () => { 
        const props = {
            rooms: [],
            addRoom:jest.fn(),
            addLike: jest.fn(),
            deleteRoom: jest.fn()
        }
        const result = shallow(<RoomsPage {...props} />);
        expect(toJson(result)).toMatchSnapshot();
        // expect(result.find("Button").exists()).toEqual(true);
        const result2 = mount(<RoomsPage {...props} />)
        expect(result2.find("Button").length).toEqual(1);
    }
    )
})