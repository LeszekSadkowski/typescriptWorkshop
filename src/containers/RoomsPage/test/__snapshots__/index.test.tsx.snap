// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rooms page container test should render given room 1`] = `
<Container
  tag="div"
>
  <div>
    salan konferancyjna
    <br />
    <input
      onChange={[Function]}
      type="text"
      value="salan konferancyjna"
    />
    <button
      onClick={[Function]}
    >
      Dodaj pokuj
    </button>
  </div>
  <Row
    tag="div"
  />
</Container>
`;
